demo/client/scripting/lib/dsl.sh

```diff
@@ -0,0 +1,107 @@
+#!/bin/bash
+#
+# Copyright Intel Corp All Rights Reserved
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+# assumes FPC_ROOT and DEMO_CLIENT_SCRIPTS_DIR is defined
+
+. "${FPC_ROOT}"/fabric/bin/lib/common_utils.sh
+
+
+CLI="${DEMO_CLIENT_SCRIPTS_DIR}/cli"
+
+
+
+
+#  Simple "DSL" to script auction scenarios
+#---------------------------------------------
+#
+# TODO (maybe)
+# - get auction status after submit and output when rounds/auction is finished?
+#
+# terms of language are
+# - scenario [<scenario-path>] if path is missing, take directory where script file is located
+# - "submit" <user> <action>
+# - "wait" <message> [<variable-for-user-input>]
+
```

Comment:
another useful thing might be a optional `sleep` after each action to slow things down if needed for "visualisation". This can be added later as a feature

Reply:
a `comment` action would be also very useful to link the lines in the console with the script while debugging os just showing line numbers?

Reply:
> another useful thing might be a optional `sleep` after each action to slow things down if needed for "visualisation". This can be added later as a feature

for non-mock you probably have to worry less about speed :-) But i guess one could add a global option/variable `delayTime` or alike and use this to add a delay for each submit (note this is a simple script file, so if you want selective delays, you can just sprinkle the script with `sleep 2` as-is)

Reply:
> a `comment` action would be also very useful to link the lines in the console with the script while debugging os just showing line numbers?

Not 100% sure i understand: do you just mean a `debug` flag or alike which triggers all actions to print more info, e.g., line numbers?  I guess this could be done using `caller` although i wouldn't have expected these scripts becoming so long that it becomes and issue.  Note you can also easily get debug info by running script as `bash -x ....` and/or `bash -vx ...` ..